pkgs/tools/networking/openconnect/default.nix

```
@@ -1,11 +1,34 @@
-{ stdenv, fetchurl, pkgconfig, openssl ? null, gnutls ? null, gmp, libxml2, stoken, zlib, fetchgit, darwin } :
+{ stdenv, fetchurl, fetchgit,
+  pkgconfig, makeWrapper,
+  openssl ? null, gnutls ? null,
+  gmp, libxml2, stoken, zlib,
+  nettools, gawk, openresolv, coreutils, gnugrep,
+  darwin } :
 
 assert (openssl != null) == (gnutls == null);
 
-let vpnc = fetchgit {
-  url = "git://git.infradead.org/users/dwmw2/vpnc-scripts.git";
-  rev = "c0122e891f7e033f35f047dad963702199d5cb9e";
-  sha256 = "11b1ls012mb704jphqxjmqrfbbhkdjb64j2q4k8wb5jmja8jnd14";
+let vpnc = stdenv.mkDerivation {
+  name = "vpnc-scripts-c0122e891f7";
+  src = fetchgit {
+    url = "git://git.infradead.org/users/dwmw2/vpnc-scripts.git";
+    rev = "c0122e891f7e033f35f047dad963702199d5cb9e";
+    sha256 = "11b1ls012mb704jphqxjmqrfbbhkdjb64j2q4k8wb5jmja8jnd14";
+  };
+
+  buildInputs = [ makeWrapper ];
+
+  installPhase = ''
+    mkdir -p $out/bin;
+    cp vpnc-script $out/bin;
+
+    substituteInPlace $out/bin/vpnc-script \
+      --replace "which" "type -P" \
+      --replace "/sbin/resolvconf" "${openresolv}/bin/resolvconf";
+
+    wrapProgram $out/bin/vpnc-script \
+      --set OS "Linux" \
+      --prefix PATH : "${stdenv.lib.makeBinPath [ nettools gawk openresolv coreutils gnugrep ]}";
```

```suggestion
    mkdir -p $out/bin
    cp vpnc-script $out/bin

    substituteInPlace $out/bin/vpnc-script \
      --replace "which" "type -P" \
      --replace "/sbin/resolvconf" "${openresolv}/bin/resolvconf"

    wrapProgram $out/bin/vpnc-script \
      --set OS "Linux" \
      --prefix PATH : "${stdenv.lib.makeBinPath [ nettools gawk openresolv coreutils gnugrep ]}"
```

What happens if OS is set to Linux on darwin?